nuitka/nodes/ImportNodes.py

```diff
@@ -1015,9 +1026,19 @@ def _attemptFollow(self, module_name):
             module_name = resolveModuleName(module_name)
 
             while True:
+                backup_module_name = module_name
                 module_name = module_name.getPackageName()
 
                 if module_name is None:
+                    self.used_modules = [
+                        ModuleUsageAttempt(
+                            backup_module_name,
+                            module_filename,
+                            self.finding,
+                            level,
+                            None,
```

Comment:
See above, setting source_ref would be very good here too. Also this is overwriting `self.used_modules` rather than adding to it. It should be a list of everything attempted during the pass, not just one.